rmgpy/solver/simple.pyx

```diff
@@ -227,27 +259,48 @@ cdef class SimpleReactor(ReactionSystem):
         cdef numpy.ndarray[numpy.float64_t, ndim=1] res, kf, kr, knet, delta, equilibriumConstants
         cdef int numCoreSpecies, numCoreReactions, numEdgeSpecies, numEdgeReactions, numPdepNetworks
         cdef int i, j, z, first, second, third
-        cdef double k, V, reactionRate
+        cdef double k, V, reactionRate, T, P, Peff
         cdef numpy.ndarray[numpy.float64_t, ndim=1] coreSpeciesConcentrations, coreSpeciesRates, coreReactionRates, edgeSpeciesRates, edgeReactionRates, networkLeakRates
-        cdef numpy.ndarray[numpy.float64_t, ndim=1] C
-        cdef numpy.ndarray[numpy.float64_t, ndim=2] jacobian, dgdk
+        cdef numpy.ndarray[numpy.float64_t, ndim=1] C, y_coreSpecies
+        cdef numpy.ndarray[numpy.float64_t, ndim=2] jacobian, dgdk, colliderEfficiencies
+        cdef numpy.ndarray[numpy.int_t, ndim=1] pdepColliderReactionIndices
+        cdef list pdepColliderKinetics
 
         ir = self.reactantIndices
         ip = self.productIndices
-        equilibriumConstants = self.equilibriumConstants
-
-        kf = self.forwardRateCoefficients
-        kr = self.reverseRateCoefficients
         
-        inet = self.networkIndices
-        knet = self.networkLeakCoefficients
-
         numCoreSpecies = len(self.coreSpeciesRates)
         numCoreReactions = len(self.coreReactionRates)
         numEdgeSpecies = len(self.edgeSpeciesRates)
         numEdgeReactions = len(self.edgeReactionRates)
         numPdepNetworks = len(self.networkLeakRates)
         
+        kf = self.forwardRateCoefficients
+        kr = self.reverseRateCoefficients
+        
+        y_coreSpecies = y[:numCoreSpecies]
+        
+        # Recalculate any forward and reverse rate coefficients that involve pdep collision efficiencies
+        if self.pdepColliderReactionIndices.shape[0] != 0:
+            T = self.T.value_si
+            P = self.P.value_si
+            equilibriumConstants = self.equilibriumConstants
+            pdepColliderReactionIndices = self.pdepColliderReactionIndices
+            pdepColliderKinetics = self.pdepColliderKinetics
+            colliderEfficiencies = self.colliderEfficiencies
+            for i in range(pdepColliderReactionIndices.shape[0]):
+                # Calculate effective pressure
+                Peff = P*numpy.sum(colliderEfficiencies[i]*y_coreSpecies / numpy.sum(y_coreSpecies)) 
+                kf[pdepColliderReactionIndices[i]] = pdepColliderKinetics[i].getRateCoefficient(T, Peff)
+                kr[pdepColliderReactionIndices[i]] = kf[pdepColliderReactionIndices[i]]/equilibriumConstants[pdepColliderReactionIndices[i]]
+                
+            # Update object's forward and reverse rate coefficients
+            self.forwardRateCoefficients = kr
+            self.reverseRateCoefficients = kf
```

Comment:
1) did you mean to switch forward and reverse here? (forward becomes kr, and reverse becomes kf)?
2) if not, I don't think these lines are needed at all - you update the arrays in place, the originals change too (`kr` is just the name of `self.reverseRateCoefficients` in the local scope, not a deep copy of it)